inputfiles/addedTypes.json

```
@@ -2368,6 +2368,35 @@
                         }
                     }
                 }
+            },
+            "ValueTypeMap": {
+                "name": "ValueTypeMap",
+                "legacy-namespace": "WebAssembly",
+                "exposed":"Window Worker Worklet",
+                "members": {
+                    "member": {
+                        "f32": {
+                            "name": "f32",
+                            "override-type": "number",
+                            "required": 1
+                        },
+                        "f64": {
+                            "name": "f64",
+                            "override-type": "number",
+                            "required": 1
+                        },
+                        "i32": {
+                            "name": "i32",
+                            "override-type": "number",
+                            "required": 1
+                        },
+                        "i64": {
+                            "name": "i64",
+                            "override-type": "bigint",
+                            "required": 1
+                        }
```

```suggestion
                        }
                        "externref": {
                            "name": "externref",
                            "override-type": "unknown",
                            "required": 1
                        },
                        "anyfunc": {
                            "name": "anyfunc",
                            "override-type": "Function",
                            "required": 1
                        },
                        "funcref": {
                            "name": "funcref",
                            "override-type": "Function",
                            "required": 1
                        },
                        "v128": {
                            "name": "v128",
                            "override-type": "never",
                            "required": 1
                        }
```
¯\\_(ツ)\_/¯ would this work?
Also, I'm not sure if you want `funcref` in there, it's only stage 2.

---

Actually, `funcref` is already the canonical name: <https://github.com/WebAssembly/reference-types/pull/85>

---

That said, I’d rather hold off on this suggestion until the **Reference Types** proposal is merged into the **WebAssembly** spec, since with this PR, it will be easy for downstream code to extend `ValueTypeMap` by using interface merging:
```ts
declare global {
	namespace WebAssembly {
		interface ValueTypeMap {
			externref: unknown;
			funcref: Function;
		}
	}
}
```

---

I see, `anyfunc` isn't even standard yet (I've been using experimental flags forever), I agree, just drop the references types, on the other hand, "v128" is much closer, do you want to keep it?